src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -0,0 +1,763 @@
+//      const auto equals_element = equals(in_expression.value(), type_compatible_elements[element_idx]);
+//      predicate_disjunction = or_(predicate_disjunction, equals_element);
+//    }
+//
+//    return evaluate_expression<int32_t>(*predicate_disjunction);
+//  } else if (right_expression.type == ExpressionType::Select) {
+//    Fail("Unsupported ExpressionType used in InExpression");
+//  } else {
+//    Fail("Unsupported ExpressionType used in InExpression");
+//  }
+//
+//  return {};
+//}
+
+
+//template<typename T>
+//std::vector<ExpressionResult<T>> ExpressionEvaluator::evaluate_array_expression(const ArrayExpression& array_expression) {
+//  const auto element_data_type = array_expression.common_element_data_type();
+//  Assert(element_data_type, "Can't evaluate Array with incompatible Element types")
+//
+//  std::vector<ExpressionResult<T>> result(array_expression.elements().size());
+//
+//  resolve_data_type(*element_data_type, [&](const auto element_data_type_t) {
+//    using ElementDataType = typename decltype(element_data_type_t)::type;
+//
+//    for (auto element_idx = size_t{})
+//  });
+//}
+
+
+template<typename R, typename Functor>
+ExpressionResult<R> ExpressionEvaluator::evaluate_binary_with_default_null_logic(const AbstractExpression& left_expression,
+                                                         const AbstractExpression& right_expression) {
+  ExpressionResult<R> result;
+
+  resolve_to_expression_results(left_expression, right_expression, [&](const auto &left, const auto &right) {
+    using LeftDataType = typename std::decay_t<decltype(left)>::Type;
+    using RightDataType = typename std::decay_t<decltype(right)>::Type;
+
+
+    if constexpr (Functor::template supports<R, LeftDataType, RightDataType>::value) {
+      const auto result_size = _result_size(left, right);
+      auto nulls = _evaluate_default_null_logic(left.nulls, right.nulls);
+
+      std::vector<R> values(result_size);
+      if (left.size() == right.size()) {
+        for (auto row_idx = ChunkOffset{0}; row_idx < result_size; ++row_idx) {
+          Functor{}(values[row_idx], left.values[row_idx], right.values[row_idx]);
+        }
+      } else if (left.size() > right.size()) {
+        for (auto row_idx = ChunkOffset{0}; row_idx < result_size; ++row_idx) {
+          Functor{}(values[row_idx], left.values[row_idx], right.values.front());
+        }
+      } else {
+        for (auto row_idx = ChunkOffset{0}; row_idx < result_size; ++row_idx) {
+          Functor{}(values[row_idx], left.values.front(), right.values[row_idx]);
+        }
+      }
+
+      result = ExpressionResult<R>{std::move(values), std::move(nulls)};
+    } else {
+      Fail("BinaryOperation not supported on the requested DataTypes");
+    }
+  });
+
+  return result;
+}
+
+template<typename R, typename Functor>
+ExpressionResult<R> ExpressionEvaluator::evaluate_binary_with_custom_null_logic(const AbstractExpression& left_expression,
```

Comment:
I would expect a method "_with_custom_null_logic" to have parameter "custom_null_logic". What is this about?